chain_of_trust.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/openpgp"
+
+	"github.com/taskcluster/taskcluster-client-go/queue"
+)
+
+type ChainOfTrust struct {
+}
+
+type ArtifactHash struct {
+	Name string `json:"name"`
+	Hash string `json:"hash"`
+}
+
+type CotExtra struct {
+	PublicIPAddress  string `json:"publicIpAddress"`
+	PrivateIPAddress string `json:"privateIpAddress"`
+	InstanceID       string `json:"instanceId"`
+	InstanceType     string `json:"instanceType"`
+	Region           string `json:"region"`
+}
+
+type ChainOfTrustCertificate struct {
+	Artifacts   []ArtifactHash               `json:"artifacts"`
+	Task        queue.TaskDefinitionResponse `json:"task"`
+	TaskID      string                       `json:"taskId"`
+	RunID       uint                         `json:"runId"`
+	WorkerGroup string                       `json:"workerGroup"`
+	WorkerID    string                       `json:"workerId"`
+	Extra       CotExtra                     `json:"extra"`
+}
+
+func (cot *ChainOfTrust) Initialise() error {
+	return nil
+}
+
+func (cot *ChainOfTrust) IsEnabled(fl EnabledFeatures) bool {
+	return fl.GenerateCertificate
+}
+
+func (cot *ChainOfTrust) Killed(task *TaskRun) error {
+	logFile := filepath.Join(TaskUser.HomeDir, "public", "logs", "live_backing.log")
+	certifiedLogFile := filepath.Join(TaskUser.HomeDir, "public", "logs", "certified.log")
+	signedCert := filepath.Join(TaskUser.HomeDir, "public", "logs", "certified.log.gpg")
```

Comment:
nit: I would do:

```go
logPath := filepath.Join(TaskUser.HomeDir, "public", "logs")
logFile := filepath.Join(logPath, "live_backing.log")
certifiedLogFile := filepath.Join(logPath, "certified.log")
signedCert := filepath.Join(logPath, "certified.log.gpg")
```

Easier to change the log path just in case.